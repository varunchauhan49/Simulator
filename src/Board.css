body {
	background: #abdc28;
	padding: 20px;
}

table {
	margin: 0 auto;
	border-collapse: collapse;
	background: black;
}
td {
    width: 50px; height: 50px;
}

tr:nth-child(odd) td:nth-child(even), tr:nth-child(even) td:nth-child(odd) {
	background: white;
}