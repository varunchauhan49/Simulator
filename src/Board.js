import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Board.css';

/*
Board class takes x,y coordinates to move robot on the board.
*/
class Board extends Component {
  constructor(props){
    super(props);
    this.state = {
      x: props.posX,
      y: props.posY,
      face: props.face
    }
  }
  componentWillReceiveProps(nextProps){
    let {posX,posY,face} = this.props;
    if(posX !== nextProps.posX){
      this.setState({x:nextProps.posX});
    }
    if(posY !== nextProps.posY){
      this.setState({y:nextProps.posY});
    }
    if(face !== nextProps.face){
      this.setState({face:nextProps.face});
    }
  }
  render() {
    const { x, y } = this.state;
    const {rows,cols} = this.props;
    let rowsItems = Array(rows).fill().map((_,i) => {
      let tds = Array(cols).fill().map((_,j) => {
        if(i === y && j === x){
          return(
            <td key={'td:' + i + ':' + j}>
              <i className="fas fa-chess-bishop" style={{color:'red',fontSize:'32px'}}></i>
            </td>
          )
        }
        else{
          return(<td key={'td:' + i + ':' + j}></td>)
        }
      })
      return (
        <tr key={"tr:" + i}>
          {tds}
        </tr>
      )
    });
    // Reverse the array
    rowsItems = rowsItems.reverse();

    return (
      <div className="Board">
        <table>
          <tbody>
            {rowsItems}
          </tbody>
        </table>
      </div>
    );
  }
}

Board.propTypes = {
  posX: PropTypes.number.isRequired,
  posY: PropTypes.number.isRequired,
  rows: PropTypes.number.isRequired,
  cols: PropTypes.number.isRequired,
  face: PropTypes.string
}

export default Board;
