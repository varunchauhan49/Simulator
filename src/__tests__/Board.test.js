import { shallow } from 'enzyme';
import Board from '../Board';
import React from 'react';

describe('Board', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Board posX={-1} posY={-1} rows={8} cols={8} face={''} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('evaluate state of class Board when data passed via prop', () => {
    const wrapper = shallow(<Board posX={-1} posY={-1} rows={8} cols={8} face={''} />);

    expect(wrapper.state().x).toBe(-1);
    expect(wrapper.state().y).toBe(-1);
    expect(wrapper.state().face).toBe('');

    wrapper.setProps({ posX: 1, posY: 1, face: 'north' });
    expect(wrapper.state().x).toBe(1);
    expect(wrapper.state().y).toBe(1);
    expect(wrapper.state().face).toBe('north');
  });
});