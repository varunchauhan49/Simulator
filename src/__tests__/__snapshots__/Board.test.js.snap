// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<div
  className="Board"
>
  <table>
    <tbody>
      <tr
        key="tr:7"
      >
        <td
          key="td:7:0"
        />
        <td
          key="td:7:1"
        />
        <td
          key="td:7:2"
        />
        <td
          key="td:7:3"
        />
        <td
          key="td:7:4"
        />
        <td
          key="td:7:5"
        />
        <td
          key="td:7:6"
        />
        <td
          key="td:7:7"
        />
      </tr>
      <tr
        key="tr:6"
      >
        <td
          key="td:6:0"
        />
        <td
          key="td:6:1"
        />
        <td
          key="td:6:2"
        />
        <td
          key="td:6:3"
        />
        <td
          key="td:6:4"
        />
        <td
          key="td:6:5"
        />
        <td
          key="td:6:6"
        />
        <td
          key="td:6:7"
        />
      </tr>
      <tr
        key="tr:5"
      >
        <td
          key="td:5:0"
        />
        <td
          key="td:5:1"
        />
        <td
          key="td:5:2"
        />
        <td
          key="td:5:3"
        />
        <td
          key="td:5:4"
        />
        <td
          key="td:5:5"
        />
        <td
          key="td:5:6"
        />
        <td
          key="td:5:7"
        />
      </tr>
      <tr
        key="tr:4"
      >
        <td
          key="td:4:0"
        />
        <td
          key="td:4:1"
        />
        <td
          key="td:4:2"
        />
        <td
          key="td:4:3"
        />
        <td
          key="td:4:4"
        />
        <td
          key="td:4:5"
        />
        <td
          key="td:4:6"
        />
        <td
          key="td:4:7"
        />
      </tr>
      <tr
        key="tr:3"
      >
        <td
          key="td:3:0"
        />
        <td
          key="td:3:1"
        />
        <td
          key="td:3:2"
        />
        <td
          key="td:3:3"
        />
        <td
          key="td:3:4"
        />
        <td
          key="td:3:5"
        />
        <td
          key="td:3:6"
        />
        <td
          key="td:3:7"
        />
      </tr>
      <tr
        key="tr:2"
      >
        <td
          key="td:2:0"
        />
        <td
          key="td:2:1"
        />
        <td
          key="td:2:2"
        />
        <td
          key="td:2:3"
        />
        <td
          key="td:2:4"
        />
        <td
          key="td:2:5"
        />
        <td
          key="td:2:6"
        />
        <td
          key="td:2:7"
        />
      </tr>
      <tr
        key="tr:1"
      >
        <td
          key="td:1:0"
        />
        <td
          key="td:1:1"
        />
        <td
          key="td:1:2"
        />
        <td
          key="td:1:3"
        />
        <td
          key="td:1:4"
        />
        <td
          key="td:1:5"
        />
        <td
          key="td:1:6"
        />
        <td
          key="td:1:7"
        />
      </tr>
      <tr
        key="tr:0"
      >
        <td
          key="td:0:0"
        />
        <td
          key="td:0:1"
        />
        <td
          key="td:0:2"
        />
        <td
          key="td:0:3"
        />
        <td
          key="td:0:4"
        />
        <td
          key="td:0:5"
        />
        <td
          key="td:0:6"
        />
        <td
          key="td:0:7"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
